tests/PHPStan/Analyser/nsrt/non-empty-string-replace-functions.php

```diff
@@ -18,7 +18,7 @@ public function replace(string $search, string $replacement, string $subject){
         assertType('non-empty-string', str_replace($search, $replacement, $subject));
         assertType('non-empty-string', str_ireplace($search, $replacement, $subject));
 
-        assertType('non-empty-string|null', preg_replace($search, $replacement, $subject));
+        assertType('(non-empty-string|null)', preg_replace($search, $replacement, $subject));
 
```

Comment:
Seems a preexisting problem, but it's wrong: `preg_replace('/^a(b?)/', '\\1', 'a');`

Reply:
Please open a new issue